tests/server/app/test_data_discovery.py

```diff
@@ -0,0 +1,72 @@
+import unittest
+
+import xcube_geodb_openeo.server.app.flask as flask_server
+import xcube_geodb_openeo.server.app.tornado as tornado_server
+import xcube_geodb_openeo.server.cli as cli
+import xcube_geodb_openeo.server.api as api
+import urllib3
+import multiprocessing
+import json
+
+from xcube_geodb_openeo.server.config import load_config
+
+
+class DataDiscoveryTest(unittest.TestCase):
+    flask = None
+    tornado = None
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        config = load_config('../../test_config.yml')
+
+        cls.flask_base_url = f'http://127.0.0.1:{cli.DEFAULT_PORT + 1}'
+
+        cls.flask = multiprocessing.Process(
+            target=flask_server.serve,
+            args=(config, '127.0.0.1', cli.DEFAULT_PORT + 1,
+                  False, False)
+        )
+        cls.flask.start()
+
+        cls.tornado_base_url = f'http://127.0.0.1:{cli.DEFAULT_PORT + 2}'
+        cls.tornado = multiprocessing.Process(
+            target=tornado_server.serve,
+            args=(config, '127.0.0.1', cli.DEFAULT_PORT + 2,
+                  False, False)
+        )
+        cls.tornado.start()
+
+        cls.http = urllib3.PoolManager()
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        cls.flask.terminate()
+        cls.tornado.terminate()
+
+    def test_collections(self):
+        for base_url in [self.flask_base_url, self.tornado_base_url]:
```

Comment:
If a tests fails, you would not know whether tornado or flask cause the failure. Suggestion is to define
```python
     msg = f"for url {base_url}"
```
and use it as custom message in the asserts calls in the for loops.

Reply:
Done.